### PR TITLE
refactor: change structure used to track unused slots in the map

### PR DESCRIPTION
**Pull request by imrn99**
# Description

Change to the following attributes of `TwoMap`:

- `free_darts: Vec<DartIdentifier>`: rename to `unused_darts` & change type to `BTreeSet`
- `free_vertices: Vec<VertexIdentifier>`: rename to `unused_vertices` & change type to `BTreeSet`

New names are less ambiguous and sets are a better fit for our use-case.

This should probably be merged after #15.

## Scope

- [x] Code: `honeycomb-core`

## Type of change

- [x] Refactor

## Necessary follow-up

...

## Mentions

...

### CODE REVIEW

File: honeycomb-core/src/twomap.rs [hunk 1]
```diff
@@ -294,15 +295,15 @@ impl<const N_MARKS: usize> TwoMap<N_MARKS> {
...
     ///
     pub fn new(n_darts: usize, n_vertices: usize) -> Self {
-        let vertices = vec![[0.0, 0.0]; n_vertices];
+        let vertices = vec![Vertex2::from([0.0, 0.0]); n_vertices];
         let betas = vec![[0; TWO_MAP_BETA]; n_darts + 1];
 
         Self {
             vertices,
-            free_vertices: Vec::with_capacity(n_vertices),
+            unused_vertices: BTreeSet::new(),
```
Comment (cedricchevalier19) [inline honeycomb-core/src/twomap.rs:303]: You loose the pre-allocation here, is it important?

File: honeycomb-core/src/twomap.rs [hunk 2]
```diff
@@ -794,7 +795,7 @@ impl<const N_MARKS: usize> TwoMap<N_MARKS> {
     /// See [TwoMap] example.
     ///
     pub fn remove_vertex(&mut self, vertex_id: VertexIdentifier) {
-        self.free_vertices.push(vertex_id);
+        assert!(self.unused_vertices.insert(vertex_id));
```
Comment (cedricchevalier19) [inline honeycomb-core/src/twomap.rs:798]: Add a comment to explain the goal of this assert, or comment the failure in the documentation above.

**Review by cedricchevalier19 — APPROVED** [review]
Comment on the speed-up of this development.

File: honeycomb-core/src/twomap.rs (hunk 2 above)
Comment (imrn99) [reply honeycomb-core/src/twomap.rs:820]: Added doc, comment and some unit tests for this in 7fafa42.

File: honeycomb-core/src/twomap.rs (hunk 1 above)
Comment (imrn99) [reply honeycomb-core/src/twomap.rs:305]: I will run benchmarks and compare 0.1.1 to 7fafa42 (which I plan on making 0.1.2 with) . I will add relevant comparisons somewhere, not sure where yet.

**Pull request merged**